src/views/signUpBuyer.jsx

```diff
@@ -0,0 +1,136 @@
+import React, { useState } from 'react';
+import { NavLink, useNavigate } from 'react-router-dom';
+import { ToastContainer, toast } from 'react-toastify';
+import CART from '../assets/CART.png';
+
+function SignUpBuyer() {
+  const navigate = useNavigate();
+  const [btn, setbtn] = useState({
+    display: 'hidden',
+    text: `Sign Up`,
+  });
+
+  const allFields = [
+    { placeh: 'enter first name ', label: ' first name', type: 'text' },
+    { placeh: 'enter last name ', label: ' last name', type: 'text' },
+    { placeh: 'enter email ', label: ' email', type: 'email' },
+    { placeh: 'phone ', label: ' phone', type: 'tel' },
+    { placeh: 'password ', label: ' password', type: 'password' },
+    {
+      placeh: 'confirm password ',
+      label: 'confirm password',
+      type: 'password',
+    },
+  ];
+  const SignUp = async (e) => {
+    e.preventDefault();
+    setbtn({
+      display: 'inline-block',
+      text: `Loading...`,
+    });
```

Comment:
Use functional updates instead of object updates. When updating state based on its previous value, it's recommended to use functional updates instead of object updates. This can help avoid issues related to stale state and ensure that the updates are based on the most recent value of the state. Here's an example of how to use a functional update to update the btn state:

```javascript
setState(prevState=> ({
  ...prevState,
  property: newValue,
  otherProperty: newValue,
  // you can modify more as you want
}));
```